### PR TITLE
Scenes: Add support for shared query results of other panel

### PR DESCRIPTION
**Pull request by torkelo**
Explores one way to support sharing query results of another panel. 

* [x] Some initial review feedback
* [x] Add tests to DashboardsLoader
* [x] Add tests for ShareQueryDataProvider 

depends on https://github.com/grafana/grafana/pull/65413

### CODE REVIEW

File: public/app/features/scenes/dashboard/ShareQueryDataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    const sourceData = source.state.$data;
+    if (!sourceData) {
+      console.log('No source data found for shared dashboard query');
+      return;
+    }
+
+    if (!sourceData?.isActive) {
+      // TODO, not sure the best way to handle this, how to handle deactivation of the source when we activated it?
+      sourceData?.activate();
```
Comment (dprokop) [inline public/app/features/scenes/dashboard/ShareQueryDataProvider.ts:57]: @torkelo I think i have faced the same issue when tried the `inputParams` approach. What I did there I had a references map in the SceneObjectBase, and only actually de-activated the object if the last reference was de-activating it.

In this case the source data is pretty much an implicit input param. Maybe we could try approaching this in a similar way too. Guess `activate` would require type change to accept a consumers key. And given that we have activation handlers now - and were thinking about making `activate` private - maybe we can change the usage of the activeate to actually support the cases like in this PR? Then we would have:
- `private activate` - for self-activation handling
- `addActivationHandler` - for objects interactions handling
- `public activateFrom(source)` - for external activation handling, subject of reference counting

Comment (dprokop) [reply public/app/features/scenes/dashboard/ShareQueryDataProvider.ts:57]: I wonder if TS allows two methods (public/ and private) with the same name but different signatture...

File: public/app/features/scenes/dashboard/utils.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { SceneDeactivationHandler, SceneObject } from '@grafana/scenes';
+
+export function getVizPanelKeyForPanelId(panelId: number) {
+  return `panel-${panelId}`;
```
Comment (ryantxu) [inline public/app/features/scenes/dashboard/utils.ts:4]: This makes me slightly nervous, but not enough to think it is a really bad idea.  In https://github.com/grafana/grafana-iot-twinmaker-app we define a complex application by dynamically replacing the layout with the layout from a set of dashbaords that depend on conditions (template variables and basic evaluations). In this scenario, we may end up with multiple VizPanels that have the same key 🤔

**Review by dprokop — APPROVED** [review]
Looks great. Don't you want to add a demo for testing the with/without transformations scenario? But may be waste of time given that those demos will sooner or later be moved to scenes epo, and sucha. demo can't live there.

File: public/app/features/scenes/dashboard/utils.ts (hunk 2 above)
Comment (torkelo) [reply public/app/features/scenes/dashboard/utils.ts:4]: @ryantxu but panel ids should be unique, right?

**Comment by torkelo** [comment]
> Looks great. Don't you want to add a demo for testing the with/without transformations scenario
@dprokop not sure we need that as we have two gdev dashboads that test this quite well:

http://localhost:3000/d/ZqZnVvFZz/datasource-tests-shared-queries?orgId=1
http://localhost:3000/d/fYGWTVW4k/transforms-reuse-dashboard-queries?orgId=1

**Comment by dprokop** [comment]
yeah, these dashboards are sufficient.

**Pull request merged**